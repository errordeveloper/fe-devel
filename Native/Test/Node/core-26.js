/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

FABRIC = require('Fabric').createClient();

var Vec2 = function( x, y ) {
  if ( typeof x === "number" && typeof y === "number" ) {
    this.x = x;
    this.y = y;
  }
  else if ( x === undefined && y === undefined ) {
    this.x = 0;
    this.y = 0;
  }
  else throw "new Vec2: invalid arguments";
};

var desc = {
  members: { x:'Scalar', y:'NotAType' },
  constructor: Vec2
};

FABRIC.RegisteredTypesManager.registerType( 'Vec2', desc );
FABRIC.flush();

FABRIC.close();
