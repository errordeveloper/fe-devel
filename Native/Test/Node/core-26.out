
node.js:201
        throw e; // process.nextTick error, or 'error' event on first tick
              ^
Fabric core exception: RT.registerType('{"name":"Vec2","members":[{"name...'): registerType('Vec2'): 'members': index 2: 'type': 'NotAType': type not registered
